graylog2-server/src/main/java/org/graylog/plugins/views/storage/migration/state/actions/MigrationActionsImpl.java

```diff
@@ -256,7 +256,11 @@ public boolean dataNodeStartupFinished() {
     @Override
     public void startRemoteReindex() {
         final String allowlist = getStateMachineContext().getActionArgumentOpt("allowlist", String.class).orElse(null);
-        final URI hostname = Objects.requireNonNull(URI.create(getStateMachineContext().getActionArgument("hostname", String.class)), "hostname has to be provided");
+        String host = getStateMachineContext().getActionArgument("hostname", String.class);
+        if (host.endsWith("/")) {
+            host = host.substring(0, host.length() - 1);
+        }
+        final URI hostname = Objects.requireNonNull(URI.create(host), "hostname has to be provided");
```

Comment:
I think the order of checks is wrong here. It's too late to require a nonnull host if we use .endsWith before.

Reply:
good point. I think `getActionArgument` will already throw a `IllegalArgumentException` if the value is missing, but I added a check for a non blank string on top.